### PR TITLE
Android sample for development

### PR DESCRIPTION
**Pull request by hannesa2**
My idea is to have a demo and a development Android project with shared code (as much as possible).
Ok, I manged this, but I failed the reduce maven and use repo plugin code with `includeBuild '../..' `

Do you have an idea how to use plugin without maven ? Previous we did it, but now I've issues

### CODE REVIEW

File: demo/androidDev/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,142 @@
+plugins {
+    id 'com.android.application'
+    id "org.jsonschema2dataclass"
```
Comment (hannesa2) [inline demo/androidDev/app/build.gradle:3]: It miss this

File: demo/androidDev/build.gradle [hunk 2]
```diff
@@ -0,0 +1,18 @@
+buildscript {
+    repositories {
+        google()
+        jcenter()
+        // maven { url "https://plugins.gradle.org/m2/" }
```
Comment (hannesa2) [inline demo/androidDev/build.gradle:5]: When I remove maven

File: demo/androidDev/build.gradle [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:4.1.3'
+        // classpath "gradle.plugin.org.jsonschema2dataclass:jsonschema2dataclass:2.1.0"
```
Comment (hannesa2) [inline demo/androidDev/build.gradle:9]: and remove this maven plugin. 
How to use checked in repository plugin ?

File: demo/androidDev/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        sourceCompatibility 1.8
+        targetCompatibility 1.8
+    }
+
+    sourceSets {
+        main {
+            manifest.srcFile '../../android/app/src/main/AndroidManifest.xml'
+            java.srcDirs = ['../../android/app/src/main/java']
+            resources.srcDirs = ['../../android/app/src/main/resources']
+            res.srcDirs = ['../../android/app/src/main/res']
```
Comment (hannesa2) [inline demo/androidDev/app/build.gradle:36]: The main idea is to use shared code

File: demo/androidDev/build.gradle (hunk 3 above)
Comment (eirnym) [reply demo/androidDev/build.gradle:9]: Only this `classpath` line breaks possibility of usage `includeBuild` in `settings.gradle`. The rest could stay the same as it is in pure demo

Comment (hannesa2) [reply demo/androidDev/build.gradle:9]: When I reintroduce this line 
`classpath "gradle.plugin.org.jsonschema2dataclass:jsonschema2dataclass:2.1.0"`
I run into 

image

I guess I need to point anyhow to `'../..'` But how ?

**Comment by hannesa2** [comment]
I've no clue what's going on here https://github.com/jsonschema2dataclass/js2d-gradle/pull/81/checks?check_run_id=2177911058

image

**Comment by eirnym** [comment]
we need to set source as `source.setFrom files(...)` rather `source = files(...)`

**Review by eirnym — APPROVED** [review]
lets's call it a day. it's finally working and I have a tendency to release soon.

**Comment by hannesa2** [comment]
Nice, you made it work !

**Comment by hannesa2** [comment]
I did a cleanup/squash of commits. Merge or not merge is up to you.

The benefit is the ability to introduce the usage (in sample) of new features immediate in same commit /branch without a new release.

**Comment by eirnym** [comment]
Thanks, setting by `setFrom` and other methods is generally better and more safe and flexible than by using `=` notation.

**Comment by eirnym** [comment]
It's a huge benefit for this project, thank you

**Pull request merged**